bin/crew

```diff
@@ -302,40 +330,71 @@ def generate_compatible
   puts 'Generating compatible packages done.'.orange if @opt_verbose
 end
 
-def search (pkgName, silent = false)
-  pkgPath = CREW_PACKAGES_PATH + pkgName + '.rb'
-  begin
-    return set_package(pkgName, pkgPath) if File.exist?(pkgPath)
-  rescue => e
-    puts "Error with #{pkgName}.rb: #{e}".lightred unless e.to_s.include?('uninitialized constant')
-  end
-  unless File.exist?(pkgPath) && silent
+def search (name, silent = false)
+  # search: Same as set_package(), but package name and component name need to be passed in a string
+  #
+  #   name: "<pkgName>::<componentName(optional)>"
+  #
+  pkgName, componentName = name.split('::', 2)
+  componentName ||= 'all'
+  pkgPath = "#{CREW_PACKAGES_PATH}/#{pkgName}.rb"
+
+  unless File.exist?(pkgPath)
     @pkg = nil
-    abort "Package #{pkgName} not found. 😞".lightred unless silent
-  return
+
+    unless silent
+      abort "Package #{pkgName} not found. 😞".lightred
+    else
+      return false
+    end
+  end
+  
+  pkgPath = CREW_PACKAGES_PATH + pkgName + '.rb'
+  set_package(pkgName, componentName)
+
+  unless @pkg.componentFileFilter.has_key?(componentName)
+    unless silent
+      abort "Component #{componentName} not found :/".lightred
+    else
+      return false
+    end
   end
 end
 
-def regexp_search(pkgPat)
-  re = Regexp.new(pkgPat, true)
-  results = Dir[CREW_PACKAGES_PATH + '*.rb'].sort \
-    .select  { |f| File.basename(f, '.rb') =~ re } \
-    .each    { |f| print_package(f, @opt_verbose) }
-  if results.empty?
-    Dir[CREW_PACKAGES_PATH + '*.rb'].each do |packagePath|
-      packageName = File.basename packagePath, '.rb'
-      begin
-        set_package packageName, packagePath
-      rescue => e
-        puts "Error with #{pkgName}.rb: #{e}".red unless e.to_s.include?('uninitialized constant')
+def regexp_search(pattern)
+  regex = Regexp.new(pattern, Regexp::IGNORECASE)
+  any_matches = nil
+
+  Dir.glob("#{CREW_PACKAGES_PATH}/*.rb") do |pkgPath|
+    set_package( File.basename(pkgPath, '.rb') )
+
+    # also check components under packages
+    @pkg.componentList.each do |componentName|
+      fullName = "#{@pkg.name}::#{componentName}"
+
+      if fullName =~ regex
+        @pkg.target_component = componentName
+        print_current_package(false)
+        any_matches ||= true
       end
-      if ( @pkg.description =~ /#{pkgPat}/i )
-        print_current_package @opt_verbose
-        results.push(packageName)
+    end
+  end
+
+  return if any_matches
+
+  Dir.glob("#{CREW_PACKAGES_PATH}/*.rb") do |pkgPath|
```

Comment:
Use `Dir[...]` instead of `Dir.glob(...)`?

Reply:
Use `Dir[...].each`?

Reply:
If there have a cleaner way to write it why not choose it :)

Reply:
All I'm saying is let's be consistent.  There are many other places in this file that use `Dir[...].each`.  Both methods are fine but if they do the same thing we should try to be consistent throughout to avoid confusion.

Reply:
Done